src/cargo/util/context/mod.rs

```diff
@@ -1546,7 +1546,10 @@ impl GlobalContext {
                 // support multiple Cargo versions at once and not
                 // get a warning.
                 let skip_warning = if let Ok(target_path) = fs::read_link(&possible) {
-                    target_path == possible_with_extension
+                    target_path == possible_with_extension ||
+                        // allow `config` -> `config.toml`, without the parent dir(s),
+                        // which is how you'd naturally do it
+                        target_path == Path::new(&filename_with_extension)
```

Comment:
is there a reason not to just make both paths absolute and compare them?

Reply:
Yes.  That is considerably slower (which may not matter very much, but this happens every time cargo starts up).

Perhaps that could be done in the  non-Unix case, but surely the perf problem is worse on Windows (and symlinks aren't really a thing there anyway).

Reply:
And yes, I had overlooked it, `same-file` should instead be used

Reply:
> And yes, I had overlooked it, same-file should instead be used

OK, thanks, willdo.